### PR TITLE
Exclude ignored measures in cnetcheck

### LINKED ISSUE #4649: cnetcheck not excluding ignored measures resulting in false positives
**ISIS version(s) affected**: All

**Description**  
cnetcheck is not properly excluding ignored measures when, at a minimum, checking for invalid latitude/longitudes and islands. These situations occur rather frequently when using feature matching, which does find points on limbs.

The net effect of this bug is reports of bad points and islands than don't actually exist.

Below are processing steps that provide the evidence of these problems, but it is in an older version of ISIS (~3.6). However, I have reviewed the code in the current version and have concluded the problem exists there as well (i.e., the code is identical in both versions).

Its possible ignored points are part of the problem as well.

**How to reproduce**  
The following processing sequence shows the steps used to demonstrate the bug. I have only included pertinent output between runs for brevity:

```
cnetcheck cnet=05_osinac_m26_registered.net fromlist= 04_network_cubes.lis prefix=05_osinac_m26_check_ ignore=true
Group = Results
  Islands       = 137
  NoLatLonCubes = 108
  LowCoverage   = 478
  SingleCube    = 5
End_Group

cnetcombinept cnetbase=05_osinac_m26_registered.net onet=05_osinac_m26_registered_clean.net imagetol=0 cleannet=true cleanmeasures=true minmeasures=0
Group = Summary
  TotalCubes        = 478
  TotalInputPoints  = 152297
  TotalOutputPoints = 62535
  PointsMerged      = 0
  PointsEvaluated   = 0
  TotalMeasures     = 536917
  MeasuresMerged    = 0
  MeasuresDeleted   = 2673285
  MinimumMeasures   = 0
End_Group

cnetcheck cnet=05_osinac_m26_registered_clean.net fromlist=04_network_cubes.lis prefix=05_osinac_m26_clean_check_ ignore=true
Group = Results
  Islands       = 19
  NoLatLonCubes = 77
  LowCoverage   = 478
  SingleCube    = 5
End_Group
```

**Possible Solution**  
There needs to be additional tests that check for ignored measures added to the lat/lon and island functions. A more thorough review of cnetcheck might be indicated as well.

Clearly you can work around this problem by cleaning first, but there are certainly cases where you may want to run additional processing to attempt to salvage the data as much as possible. Its still a bug, nonetheless.

**Additional context**  
None

### PR DESCRIPTION
**Pull request by AustinSanders**
The 'ignore' flag was previously being used only to exclude ignored points, not ignored measures.  This PR addresses this by excluding ignored measures.

## Description
Makes use of the "GetValidMeasuresInCube" function to exclude ignored measures when the "ignore" flag is set.

## Related Issue
Closes #4649 

## Motivation and Context
Including ignored measures results in false positives.

## How Has This Been Tested?
Wrote a new test that adds an ignored control measure.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/control/apps/cnetcheck/cnetcheck.cpp [hunk 1]
```diff
@@ -635,7 +635,12 @@ namespace Isis {
           createdCamera = false;
         }
 
-        QList< ControlMeasure * > measures = cnet.GetMeasuresInCube(serialNumber);
+        QList< ControlMeasure * > measures;
+        if (ignore) {
+          measures = cnet.GetValidMeasuresInCube(serialNumber);
+        } else {
+          measures = cnet.GetMeasuresInCube(serialNumber);
+        }
```
Comment (Kelvinrr) [inline isis/src/control/apps/cnetcheck/cnetcheck.cpp:643]: Should we also ignore the point if all measures are ignored? That is, can all the measures be set to ignored but not the point? If so, in that case also ignore the point automatically? Or does cnetcheck already report points with no measures?

File: isis/tests/FunctionalTestsCnetcheck.cpp [hunk 2]
```diff
@@ -146,3 +146,57 @@ TEST_F(ThreeImageNetwork, FunctionalTestCnetcheckIgnore) {
...
+  m1->SetAprioriLine(481);
+  m1->SetAprioriSample(481);
+  m1->SetCamera(cube1->camera());
+  m1->SetCubeSerialNumber(cube1Serial);
+
+  ControlMeasure *m2 = new ControlMeasure();
+  m2->SetAprioriLine(999);
+  m2->SetAprioriSample(999);
+  m2->SetCamera(cube2->camera());
+  m2->SetCubeSerialNumber(cube2Serial);
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsCnetcheck.cpp:169]: I would also set these measures to ignored and look for 0 no-lat/lon cubes. I don't think that the third measure changes the cnetcheck output.

**Review by jessemapel — COMMENTED** [review]
Test also needs to check for an island that is connected by an ignored point.

**Comment by jessemapel** [comment]
Needs a changelog entry

**Review by jessemapel — APPROVED** [review]

**Pull request merged**